### PR TITLE
Request test

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
@DavidKang, @Ana06, @evanrolfe Here is what we did on Tuesday

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 1]
```diff
@@ -36,45 +35,55 @@
...
+      end
     end
 
-    it 'shows a hint to project maintainers when there are package maintainers' do
-      login receiver
-
-      create_submit_request
+    context 'if request does not exist' do
+      it 'redirects to root_path' do
+        get :show, params: { number: '200000' }
```
Comment (DavidKang) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: This should be in a `before` block, right?

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 2]
```diff
@@ -36,45 +35,55 @@
   it { is_expected.to use_before_action(:require_request) }
 
   describe 'GET show' do
-    it 'is successful as nobody' do
-      get :show, params: { number: bs_request.number }
-      expect(response).to have_http_status(:success)
-    end
+    context 'if successful' do
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: I would say the context is 
```
context 'as nobody' do
    it 'has success status code' ...
    it 'assigns @bs_request' ...
end
```

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 3]
```diff
@@ -36,45 +35,55 @@
...
 
-    it 'shows a hint to project maintainers when there are package maintainers' do
-      login receiver
-
-      create_submit_request
+    context 'if request does not exist' do
+      it 'redirects to root_path' do
+        get :show, params: { number: '200000' }
+        expect(flash[:error]).to eq("Can't find request 200000")
+        expect(response).to redirect_to(user_show_path(User.current))
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: [nitpick] I would separate the two asserts

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 4]
```diff
@@ -36,45 +35,55 @@
...
+      before do
+        login receiver
+        bs_request.bs_request_actions.delete_all
+        bs_request_submit_action
+        get :show, params: { number: bs_request.number }
+      end
 
-      expect(assigns(:show_project_maintainer_hint)).to eq(true)
+      it 'shows a hint to project maintainers' do
+        expect(assigns(:show_project_maintainer_hint)).to eq(true)
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: [nitpick]
``
to be_true
``

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 5]
```diff
@@ -36,45 +35,55 @@
...
+    context 'when there are no package maintainers' do
+      before do
+        login receiver
+        bs_request_submit_action
+        get :show, params: { number: bs_request.number }
+      end
 
-      expect(assigns(:show_project_maintainer_hint)).to eq(false)
+      it 'does not show a hint to project maintainers by default' do
+        expect(assigns(:show_project_maintainer_hint)).to eq(false)
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: [nitpick]
```
to be_false
```

**Comment by bgeuken** [comment]
@DavidKang @ChrisBr Thanks for the review. I've updated the code now. Please have another look.

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 6]
```diff
@@ -36,45 +35,56 @@
...
+    context 'as nobody' do
+      before do
+        get :show, params: { number: bs_request.number }
+      end
 
-    it 'assigns @bs_request' do
-      get :show, params: { number: bs_request.number }
-      expect(assigns(:bs_request)).to eq(bs_request)
-    end
+      it 'respongs successfully' do
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: ``s/respongs/responds/``

**Comment by DavidKang** [comment]
@bgeuken, please rebase 😸

**Pull request merged**